skt/runner.py

```diff
@@ -291,12 +331,11 @@ def __getresults(self):
             for recipe_set_id in recipe_sets:
                 results = self.recipe_set_results[recipe_set_id]
                 for recipe_result in results.findall('.//recipe'):
-                    if recipe_result.attrib.get('result') != 'Pass':
-                        ret = self.decide_run_result_by_task(recipe_result)
+                    ret = self.decide_run_result_by_task(recipe_result)
 
-                        if ret != SKT_SUCCESS:
-                            logging.info('Failure in a recipe detected!')
-                            return ret
+                    if ret != SKT_SUCCESS:
+                        logging.info('Failure in a recipe detected!')
+                        return ret
```

Comment:
This wouldn't change the issue we were discussing last friday on IRC about returning the first error that was found on the recipes.

Maybe errors should be saved and return the .. highest value? .

Reply:
@inakimalerba it does not -- the PR just tries to simplify retcode checking here so the fix for the issue you mention is easier to manage